Dockerfile

```diff
@@ -1,10 +1,34 @@
-# Goreleaser builds a docker image according to this Dockerfile by copying the sysl binary 
-# to a golang:1.13-alpine image and setting up the entrypoint.
+# Builds a docker image according to this Dockerfile by building the sysl binary
+# on Go 1.13 (by default) and then copying it
+# to a debian:buster-slim image and setting up the entrypoint.
 #
 # The produced image is published to https://hub.docker.com/r/anzbank/sysl
+#
+# To use this Docker file run the following command
+#    docker image build --build-arg sysl_tag=v0.7.1  -t my-sysl:0.7.1 .
+
+ARG go_ver=1.13
+ARG alpine_ver=3.11
+
+FROM golang:${go_ver}-alpine${alpine_ver} as builder
+
+ARG sysl_tag
+
+RUN test -n "${sysl_tag}" || ( echo Plese set sysl_tag build-arg && exit 1 )
 
-FROM golang:1.13-alpine
+RUN apk --no-cache add git
 
-COPY sysl /
+WORKDIR /sysl
+
+RUN git clone --branch ${sysl_tag} --single-branch --depth 1 https://github.com/anz-bank/sysl.git . && \
```

Comment:
I don't know what else to say. I feel this is 100% wrong. I want `docker build .` to build what is in `.` just like every other docker-build-from-source dockerfile I have does. I need to be able to docker build my local source code before I raise a PR so I can verify it works properly in a docker image. Being only able to build a docker image of the official source code for which there is a git ref (branch/tag) is just too constraining.